Blockhandler removal, Additional Event Coverage, 0xMarketplaceZap

GundamDweeb opened the pull request:
Hosted Service - https://thegraph.com/hosted-service/subgraph/gundamdweeb/nftx-v2

 - Removed `blockHandler` and handler implementation (and related unused functions )
 - Remove `VaultDayData` & `VaultHourData` entities
 - `Withdraw` & `Deposit` event added to InventoryStaking

- [x] Update `Goerli` config & ABIs
- [x] Implement `Withdraw` and `Deposit`
- [x] Continuing with Updating `LPStaking` ABI to get the `Deposit` event added
- [x] Add 0xMarketplaceZap implementation, follow other Zap implementation & Hackmd v2.1 Doc

justincavery commented on subgraph.template.yaml:
```diff
@@ -213,6 +213,34 @@ dataSources:
           handler: handleSwapZap
+  - kind: ethereum
+    name: ZeroXMarketplaceZap
+    network: goerli
```
should this be `network: {{network}}' instead?

GundamDweeb replied:
ty, have fixed

GundamDweeb commented:
### Changeset

1. Added ZeroXMarketplaceZap to configs and implemented handlers for Buy, Swap, Sell zaps.
2. Implemented `Deposit` event on LPStaking, handler as `handleLPDeposit`
3. Updated `getMint`, `getSwap`, `getRedeem` to have default source parameter.

@justincavery On point 3, this is something we can use to add more info about where the tx is coming from, as requested in hackmd.

GundamDweeb commented:
1. Add source to Withdrawal, LPWithdrawal, Deposit, LPDeposit
2. Add Event Types to ActivityEvents (Withdrawal, LPWithdrawal, Deposit, LPDeposit)
3. Add Vault to Deposit & Withdrawal Entities
4. Add their Types to Mint,Redeem,Swap, Deposit, Withdrawal (Deposits & Withdrawal can have 2 types respectively, LPDeposit & Deposit etc)
5. Add differentiation between Zap interactions and Mint/Redeem/Swap. (ZapSwap, ZapBuy, ZapSell)

```
enum ActivityEventType {
  Swap
  Redeem
  Mint
  Deposit
  Withdrawal
  LPDeposit
  LPWithdrawal
  ZapSell
  ZapSwap
  ZapBuy
}
```

GundamDweeb commented:
Deposit, Withdrawal, Mint, Redeem & Swap entities have new ID structure.
`ID = "{entityName}-{txHash}"` e.g. `MINT-0xfe4n439g4msdx....`

GundamDweeb commented:
Added `VaultCreated`, `VaultPublished` and `UnstakeInventory` as ActivityEventTypes.

1. Redeems and/or Withdrawals that happen through the `NFTXUnstakingInventoryZap` will be tagged as type `UnstakeInventory`. No entity was added for UnstakeInventory.
2. `VaultCreated` & `VaultPublished` exist as their own entities with the corresponding ActivityEventType set.

justincavery commented:
This has been deployed across Goerli, Mainnet, and Arbitrum.
